commands/bootstrap_cli.py

```diff
@@ -278,18 +279,26 @@ def copy_secrets(project_profile, source_environment, target_environment):
 
     for secret in secrets:
         secret_name = secret[0].replace(f"/{source_environment}/", f"/{target_environment}/")
-        set_ssm_param(
-            config["app"],
-            target_environment,
-            secret_name,
-            secret[1],
-            True,
-            True,
-            f"Copied from {source_environment} environment.",
-        )
 
         click.echo(secret_name)
 
+        try:
+            set_ssm_param(
+                config["app"],
+                target_environment,
+                secret_name,
+                secret[1],
+                False,
+                False,
+                f"Copied from {source_environment} environment.",
+            )
+        except ClientError as e:
```

Comment:
It could be worth adding the exception handling logic to the set_ssm_param function and raising something more specific?
```
raise ParameterExists("xxx")
```

It might save some duplication going forward.

Also don't you want to re-raise in an else statement?

Reply:
I was thinking about adding this to the `set_ssm_param` method actually, but that entire method is not tested ATM. Left it as a question in the description of this PR.

Not sure I want to re-raise. Is there any reason to do this?

Reply:
I think re-raising the exception if the code is not `ParameterAlreadyExists` would be correct.

Reply:
We are just using `set_ssm_param` here, I would ticket it up to get that under test separately so we don't hold up getting this value add published.

Reply:
That makes sense!

Resolved in https://github.com/uktrade/copilot-tools/pull/121/commits/19403a3365f839cdea15e47840ae15cd74e8dc03.

Reply:
Created a ticket to add unit tests for the `set_ssm_param` utils method here -> https://uktrade.atlassian.net/browse/DBTP-352.